oceanmodes/baroclinic.py

```diff
@@ -117,6 +117,8 @@ def _neutral_modes_from_N2_profile_raw(z, N2, f0, depth=None, **kwargs):
     # make sure z is increasing
     if not np.all(dzc > 0):
         raise ValueError('z should be monotonically increasing')
+    if z[1]==0:
+        raise ValueError('z[0] should not be 0, as it is the depth where N2 (a gradient quantity) is defined')
```

Comment:
Why are you checking `z[1]` but raising an error about `z[0]`. Seems inconsistent.